### DISCUSSION TITLE
Using nocodb with existing sqlite database

### DISCUSSION THREAD

**Discussion by noreff** [nocodb/nocodb#877] [Q&A]
I'm trying to use nocodb with existing sqlite database but can't see any columns. Is that supported?

**Comment by o1lab** [comment]
Yes, its supported.

Any error logs in browser console or server ?

**Reply by noreff** [reply]
What I'm doing:

I'm starting a docker with a mount to /db folder on my host

```
docker run --name nocodb -v /db:/usr/app/data/ -p 8080:8080 nocodb/nocodb:latest
```

Inside that db folder I have a file test.db which is sqlite3 database.

Then I'm opening  dashboard at port 8080 and creating a project from existing database. 

![image](https://user-images.githubusercontent.com/5099618/146582086-48041eae-76db-4363-a754-0dadd280f04b.png)

I specify this path to db: `/test.db`. I also tried those with the same result: `test.db`, `/usr/app/data/test.db`

My test.db has one table inside called test.

![image](https://user-images.githubusercontent.com/5099618/146582619-91818f45-e909-4250-b47d-913d54dc8d7b.png)

That's what I expect to see in the dashboard, but that's not the case unfortunately. 

![image](https://user-images.githubusercontent.com/5099618/146582457-e0bf37ae-ecdf-4241-a2a8-999dffc3c529.png)

When I'm trying external aurora cluster everything works like a charm which make me wonder if this a bug or I'm doing something wrong? Can you please point me to my mistake if that's the case?

By the way I'm using your tool since xgencode, thank you a lot, that's a great thing.

**Comment by noreff** [comment]
Looks like I did something wrong previously, now it seems to work. Path that I end up using is `/usr/app/data/test/test.db` and file on my host is located in `/db/test/test.db`

**Comment by noreff** [comment]
UPD: I see what I did different before: I tried to connect to my existing DB which is quite big and complicated. With a test.db it works without issues, but with mine 'live' db it's giving a following error: 
```
Creating REST APIs sqlite3 - > /tt/tt__noco_1
TypeError: Cannot read property 'split' of null
    at Object.camelize (/usr/src/app/node_modules/inflection/lib/inflection.js:748:26)
    at /usr/src/app/docker/main.js:2:188341
    at Array.reduce (<anonymous>)
    at M.getInflectedName (/usr/src/app/docker/main.js:2:188223)
    at M.getColumnNameAlias (/usr/src/app/docker/main.js:2:199476)
    at M.relationsSyncAndGet (/usr/src/app/docker/main.js:2:194891)
    at async M.xcTablesPopulate (/usr/src/app/docker/main.js:2:673647)
    at async M.loadRoutes (/usr/src/app/docker/main.js:2:669028)
    at async M.init (/usr/src/app/docker/main.js:2:668593)
    at async n.init (/usr/src/app/docker/main.js:2:148748)
```
I will let you know if I found what the reason for that.

**Comment by o1lab** [comment]
Thank you for all the notes

**Reply by MichaelAntonFischer** [reply]
Hi o1lab,

I am trying to add an existing database like described in this issue, but I am not even seeing the option to create from existing database, when creating a project.
Was this removed?

**Reply by bobolive** [reply]
Same problem here....

**Reply by o1lab** [reply]
You can always refer to [latest docs](https://docs.nocodb.com/setup-and-usages/dashboard#connecting-to-external-database) :)

**Reply by bobolive** [reply]
> You can always refer to [latest docs](https://docs.nocodb.com/setup-and-usages/dashboard#connecting-to-external-database) :)

Thx!
